packages/squiggle-lang/__tests__/Reducer/Reducer_Peggy/Reducer_Peggy_ToExpression_type_test.res

```diff
@@ -0,0 +1,99 @@
+open Jest
+open Reducer_Peggy_TestHelpers
+
+describe("Peggy Types to Expression", () => {
+  describe("type of", () => {
+    testToExpression(
+      "p: number",
+      "{(:$_typeOf_$ :p #number)}",
+      ~v="{_typeReferences_: {p: #number}}",
+      (),
+    )
+  })
+  describe("type alias", () => {
+    testToExpression(
+      "type index=number",
+      "{(:$_typeAlias_$ #index #number)}",
+      ~v="{_typeAliases_: {index: #number}}",
+      (),
+    )
+  })
+  describe("type or", () => {
+    testToExpression(
+      "answer: number|string|distribution",
+      "{(:$_typeOf_$ :answer (:$_typeOr_$ (:$_constructArray_$ (#number #string #distribution))))}",
+      ~v="{_typeReferences_: {answer: {typeTag: 'typeOr',typeOr: [#number,#string,#distribution]}}}",
+      (),
+    )
+  })
+  describe("type function", () => {
+    testToExpression(
+      "f: number=>number=>number",
+      "{(:$_typeOf_$ :f (:$_typeFunction_$ (:$_constructArray_$ (#number #number #number))))}",
+      ~v="{_typeReferences_: {f: {typeTag: 'typeFunction',inputs: [#number,#number],output: #number}}}",
+      (),
+    )
+    testToExpression(
+      "f: number=>number",
+      "{(:$_typeOf_$ :f (:$_typeFunction_$ (:$_constructArray_$ (#number #number))))}",
+      ~v="{_typeReferences_: {f: {typeTag: 'typeFunction',inputs: [#number],output: #number}}}",
+      (),
+    )
+  })
+  describe("high priority modifier", () => {
+    testToExpression(
+      "answer: number<-min(1)<-max(100)|string",
```

Comment:
small point, but if the min is a number, I imagine we could infer that it's a number. Could be more work though, not needed now.

Reply:
modifiers are mapped to polymorphic functions. If there is an implementation for that type then  that works. Otherwise you get the usual function not found complaint.  Therefore the parser is not concerned with the modifier parameters.

So if you wish you can add a min, max implementation for other types (if it makes sense).  

You can define other constraints by creating the corresponding functions for the corresponding type.

Reply:
I was discussing cases where min() has a number in it. Like,

```
  "answer: <-min(1)<-max(100)|string", 
```

In this case, it should be obvious the first type is numeric, though it's not clear if its an int or float. 

Again, this is a small deal now.